src/webots/gui/WbStreamingServer.cpp

```diff
@@ -552,21 +552,28 @@ void WbStreamingServer::propagateNodeAddition(WbNode *node) {
     connectNewRobot(robot);
 }
 
-QString WbStreamingServer::simulationStateString() {
+QString WbStreamingServer::simulationStateString(bool pauseTime) {
+  QString string;
   switch (WbSimulationState::instance()->mode()) {
     case WbSimulationState::PAUSE:
-      return "pause";
+      string = pauseTime ? QString("pause: %1").arg(WbSimulationState::instance()->time()) : "pause";
+      break;
     case WbSimulationState::STEP:
-      return "step";
+      string = "step";
+      break;
     case WbSimulationState::REALTIME:
-      return "real-time";
+      string = "real-time";
+      break;
     case WbSimulationState::RUN:
-      return "run";
+      string = "run";
+      break;
     case WbSimulationState::FAST:
-      return "fast";
+      string = "fast";
+      break;
     default:
-      return QString();
+      break;
   }
+  return string;
 }
```

Comment:
Why is it required to store the returned value in a string?

Reply:
It's not, I will revert this...